### DISCUSSION TITLE
why do quick fixes import super and not crate?

### DISCUSSION THREAD

**Discussion by evbo** [rust-lang/rust-analyzer#18520] [General]
back in the day missing import would suggest importing from `crate`, now it suggests `super`. How can I make it prefer `crate`?

**Comment by lnicola** [comment]
`"rust-analyzer.imports.prefix": "crate"`

**Reply by gibfahn** [reply]
This seems to work for the "Import `crate::path::to::item`", but not for the "Qualify as `super::to::item`" one. I don't see another option, unsure if this is something worth filing an issue for.

**Reply by Veykril** [reply]
Qualify uses a different scheme, I believe that is intentional in that we want to pick smaller paths if possible (so it tries to use relative paths)

**Comment by evbo** [comment]
@lnicola thank you, this should be the default though, am I right? Other than unit tests `super` imports are wicked unpopular.

**Reply by lnicola** [reply]
I don't have a strong opinion. rust-analyzer itself prefers `crate`, but I remember noticing a while back that it didn't seem to match what the other projects were using at the time. And you're probably the first to complain about this.

**Reply by evbo** [reply]
Okay let me try another angle: isn't it more consistent to follow rust-analyzer defaults (preferring `crate`)? Isn't that the expected behavior?

**Reply by Veykril** [reply]
The expected behavior is highlight subjective, the default setting for this is currently `plain` which allows up to one `super` being emitted (if `super::super` is required it will instead pick a `crate::` path). The default should be what the majority expects but thats difficult to figure out. I do agree though that we should probably just put this to `crate` as it is the least messy of the three.

**Reply by gibfahn** [reply]
Personally I'd agree that `crate` would be a better default than `super`, but I can also see that it might be an unpopular breaking change.
